autokeras/auto/processor.py

```diff
@@ -76,3 +77,119 @@ def transform(self, data):
         # channel-wise normalize the image
         data = (data - self.mean) / self.std
         return data
+
+    def get_min_and_max(value, name):
+        if isinstance(value, (tuple, list)):
+            if len(value) != 2:
+                raise ValueError(
+                    'Argument %s expected either a float between 0 and 1, '
+                    'or a tuple of 2 floats between 0 and 1, '
+                    'but got: %s' % (value, name))
+            min_value = value[0]
+            max_value = value[1]
+        else:
+            min_value = 1. - value
+            max_value = 1. + value
+        return min_value, max_value
+
+    def augment_image(self,
+                      x_train,
+                      rotation_range=0,  # either 0, 90, 180, 270
+                      random_crop_height=0,  # fraction 0-1
+                      random_crop_width=0,  # fraction 0-1
+                      random_crop_seed=0,   # positice number
+                      brightness_range=0.,  # fraction 0-1  [X]
+                      saturation_range=0.,  # fraction 0-1  [X]
+                      contrast_range=0.,  # fraction 0-1  [X]
+                      horizontal_flip=False,  # boolean  [X]
+                      vertical_flip=False,
+                      translation_top=0,  # translation_top are the number of pixels.
+                      translation_bottom=0,
+                      translation_left=0,
+                      translation_right=0,
+                      gaussian_noise=False):  # boolean  [X]
+        x_train = tf.convert_to_tensor(x_train)
+        length_dim = len(x_train.shape)
+        if length_dim != 4:
+            raise ValueError(
+                             'The input of x_train should be a [batch_size, height, '
+                             'width, channel] '
+                             'shape tensor or list, but we get %s' % x_train.shape)
+        batch_num = x_train.shape[0]
+        target_height = x_train.shape[1]
+        target_width = x_train.shape[2]
+        channels = x_train.shape[3]
```

Comment:
Can you do it in one liner, `batch_num, target_height , target_width, channels = x_train.shape`?